### PR TITLE
[#1824] fix(trino-connector) Fix query on mysql when join pushdown was applied

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

Query failed on a MySQL JOIN operation when pushdown was applied

### Why are the changes needed?

Removes GravitinoColumnHandle to ensure that the ColumnHandle used by GravitionMetadata is JDBCColumnHandle, thereby avoiding type conversions and scenarios where conversion is not possible.

Fix: #1824

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoColumnHandle.java [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.base.Preconditions;
-import io.trino.spi.connector.ColumnHandle;
-
-/**
- * The GravitinoColumnHandle is used to transform column information between Trino and Gravitino, as
- * well as to wrap the inner connector column handle for data access.
- */
-public final class GravitinoColumnHandle implements ColumnHandle {
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoColumnHandle.java:18]: Why are we removing this class? you'd better add some description about this point.

**Comment by yuqi1129** [comment]
Please explain the causes of problem here, I noticed that it existed for a long time.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoColumnHandle.java (hunk 1 above)
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoColumnHandle.java:18]: The query optimizer iterates continuously during the query optimization process. Initially, it invokes the GravitionMetadata interface using GravitinoTableHandle and GravitinoColumnHandle as parameters. After processing by the internal connector, the returned optimization results may include JDBCTableHandle and JDBCColumnHandle. As the optimization continues, the parameters passed to the GravitionMetadata interface could be a mix of GravitinoTableHandle and JDBCColumnHandle, or JDBCTableHandle and GravitinoColumnHandle, or even just JDBCTableHandle and JDBCColumnHandle. Therefore, the previous code contained a significant amount of logic for checking the types of TableHandle and ColumnHandle. This update removes GravitinoColumnHandle to ensure that the ColumnHandle used by GravitionMetadata is JDBCColumnHandle, thereby avoiding type conversions and scenarios where conversion is not possible.

Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoColumnHandle.java:18]: > This update removes GravitinoColumnHandle to ensure that the ColumnHandle used by GravitionMetadata is JDBCColumnHandle

So, when we query tables backed by Iceberg catalogs,  the implementation of `ColumnHandle` is also `JDBCColumnHandle`?

Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoColumnHandle.java:18]: No use mysql connector is the `JDBCColumnHandle`

**Review by yuqi1129 — APPROVED** [review]
LGTM.

**Pull request merged**